nhrpd/zbuf.c

```diff
@@ -230,3 +230,15 @@ void zbuf_copy(struct zbuf *zdst, struct zbuf *zsrc, size_t len)
 		return;
 	memcpy(dst, src, len);
 }
+
+void zbuf_copy_nopull(struct zbuf *zdst, struct zbuf *zsrc, size_t len)
+{
+	const void *src;
+	void *dst;
+
+	dst = zbuf_pushn(zdst, len);
+	src = zbuf_pulln(zsrc, 0);
+	if (!dst || !src)
+		return;
+	memcpy(dst, src, len);
+}
```

Comment:
This function is identical to zbuf_copy(), an unnecessary redundancy.  Can you please just use zbuf_copy(), extend with an extra argument to control if you want to pull or not.

Reply:
I suppose you want that zsrc not be modified.
with zbuf_copy, the head pointer of zsrc is modified.
what about naming zbuf_copy_peek(dst,src, len) ?
inside, you will not modify head pointer. right?

Reply:
Adding an extra argument is a lot more change since this is used many places. I have renamed the function as suggested by @pguibert6WIND

Reply:
That works.